### PR TITLE
add aten::group_norm support

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

File: src/frontends/pytorch/src/op/group_norm.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    auto num_groups = context.const_input<int64_t>(1);
+    auto eps = context.const_input<double>(4);
+    auto input_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(data, element::i64));
+    auto shape = context.mark_node(
+        std::make_shared<opset8::Constant>(element::i64, Shape({3}), std::vector<int64_t>{0, num_groups, -1}));
+    auto reshaped_input = context.mark_node(std::make_shared<opset8::Reshape>(data, shape, true));
+    auto value_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(reshaped_input, element::i64));
+    const auto reduction_axes = std::make_shared<opset8::Range>(
+        opset8::Constant::create(element::i64, {}, {2}),
+        std::make_shared<opset8::Squeeze>(std::make_shared<opset8::ShapeOf>(value_shape, element::i64)),
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/group_norm.cpp]: I would prefer this to be done in separate lines. Also, you need to call `context.mark_node` for each node you create

Comment (mvafin) [inline src/frontends/pytorch/src/op/group_norm.cpp]: `ShapeOf(value_shape)` is always 3, right? Which means `reduction_axes` are always `[2]`, isn't it?

File: src/frontends/pytorch/src/op/group_norm.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        opset8::Constant::create(element::i64, {}, {1}),
+        element::i64);
+    auto reshaped_norm = context.mark_node(
+        std::make_shared<opset8::MVN>(reshaped_input, reduction_axes, true, eps, ov::op::MVNEpsMode::INSIDE_SQRT));
+    auto norm = context.mark_node(std::make_shared<opset8::Reshape>(reshaped_norm, input_shape, true));
+    auto input_rank = std::make_shared<opset8::Squeeze>(std::make_shared<opset8::ShapeOf>(input_shape, element::i64));
+    auto skip_last = std::make_shared<opset8::Subtract>(input_rank, opset8::Constant::create(element::i64, {}, {1}));
+    auto axes = std::make_shared<opset8::Range>(opset8::Constant::create(element::i64, {}, {1}),
+                                                skip_last,
+                                                opset8::Constant::create(element::i64, {}, {1}),
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/group_norm.cpp]: You have at least 4 constants with value 1. I think it would be better to create just one and reuse it. That is the approach which was used in tensorflow frontend. I am not fully sure if it will have any advantage, but in cases like this one it seems reasonable.

File: src/frontends/pytorch/src/op/group_norm.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                                                element::i64);
+    if (!context.input_is_none(2)) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/group_norm.cpp:33]: It would be nice to have a comments in the translator about extra inputs 2 and 3.

File: src/frontends/pytorch/src/op/group_norm.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/group_norm.cpp]: this line replaced with constant, `context.mark_node` added everywhere

File: src/frontends/pytorch/src/op/group_norm.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/group_norm.cpp]: constant reused

File: src/frontends/pytorch/src/op/group_norm.cpp (hunk 3 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/group_norm.cpp:33]: added

File: tests/layer_tests/pytorch_tests/test_group_norm.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from pytorch_layer_test_class import PytorchLayerTest
+
+
+class TestGroupNorm(PytorchLayerTest):
+    def _prepare_input(self):
+        import numpy as np
+        return (np.random.randn(20, 6, 10, 10).astype(np.float32),)
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_group_norm.py]: Could you also add tests for 3D and 5D inputs?

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/group_norm.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/group_norm.cpp]: replaced with constant

File: tests/layer_tests/pytorch_tests/test_group_norm.py (hunk 4 above)
Comment (eaidova) [reply tests/layer_tests/pytorch_tests/test_group_norm.py]: added

File: src/frontends/pytorch/src/op/group_norm.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    auto data = context.get_input(0);
+    auto num_groups = context.const_input<int64_t>(1);
+    // input 2 - weights and input 3 - bias are optional without default value, we handle them later
+    auto eps = context.const_input<double>(4);
+    auto input_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(data, element::i64));
+    auto scalar_one = context.mark_node(opset8::Constant::create(element::i64, {}, {1}));
+    auto shape = context.mark_node(
+        std::make_shared<opset8::Constant>(element::i64, Shape({3}), std::vector<int64_t>{0, num_groups, -1}));
+    auto reshaped_input = context.mark_node(std::make_shared<opset8::Reshape>(data, shape, true));
+    auto reduction_axes = context.mark_node(opset8::Constant::create(element::i64, Shape({1}), std::vector<int64_t>(1, 2)));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/group_norm.cpp]: Please fix this codestyle problem

Comment (eaidova) [reply src/frontends/pytorch/src/op/group_norm.cpp]: fixed

**Review by mvafin — APPROVED** [review]

File: src/frontends/pytorch/src/op/group_norm.cpp [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    auto reshaped_input = context.mark_node(std::make_shared<opset8::Reshape>(data, shape, true));
+    auto reduction_axes =
+        context.mark_node(opset8::Constant::create(element::i64, Shape({1}), std::vector<int64_t>(1, 2)));
+    auto reshaped_norm = context.mark_node(
+        std::make_shared<opset8::MVN>(reshaped_input, reduction_axes, true, eps, ov::op::MVNEpsMode::INSIDE_SQRT));
+    auto norm = context.mark_node(std::make_shared<opset8::Reshape>(reshaped_norm, input_shape, true));
+    auto input_rank2d = context.mark_node(std::make_shared<opset8::ShapeOf>(input_shape, element::i64));
+    auto input_rank = context.mark_node(std::make_shared<opset8::Squeeze>(input_rank2d));
+    auto skip_last = context.mark_node(std::make_shared<opset8::Subtract>(input_rank, scalar_one));
+    auto axes = context.mark_node(std::make_shared<opset8::Range>(scalar_one, skip_last, scalar_one, element::i64));
```
Comment (slyalin) [inline src/frontends/pytorch/src/op/group_norm.cpp:32]: I will have an epileptic seizure from these `mark_node`s everywhere. We should definitely do something with it in prod.
